Dispatcher/Controllers/SearchViewController.swift

```diff
@@ -0,0 +1,164 @@
+import UIKit
+
+enum iconImageName {
+    case search
+    case remove
+}
+
+class SearchViewController: UIViewController {
+    
+    @IBOutlet var contentView: UIView!
+    @IBOutlet weak var searchView: UIView!
+    @IBOutlet weak var searchTextField: UITextField!
+    @IBOutlet weak var searchClearIcon: UIImageView!
+    @IBOutlet weak var goBackButton: UIImageView!
+    
+    
+    @IBOutlet weak var recentSearchesView: UIView!
+    @IBOutlet weak var recentSearchesTableView: UITableView!
+    
+    
+    
+    var searchClearImageName: iconImageName = .search
+    var recentSearchesArray: [String] = [
+        "crypto",
+        "soccer",
+        "football"
+    ]
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchTextField.delegate = self
+        
+        searchClearIcon.addGestureRecognizer(UITapGestureRecognizer(target: searchClearIcon, action: #selector(searchClearButtonPressed)))
+        searchClearIcon.isUserInteractionEnabled = true
+        let tapGestureRecognizer1 = UITapGestureRecognizer(target: self, action: #selector(searchClearButtonPressed(tapGestureRecognizer:)))
+        searchClearIcon.addGestureRecognizer(tapGestureRecognizer1)
+        
+        goBackButton.addGestureRecognizer(UITapGestureRecognizer(target: goBackButton, action: #selector(goBackButtonPressed)))
+        goBackButton.isUserInteractionEnabled = true
+        let tapGestureRecognizer2 = UITapGestureRecognizer(target: self, action: #selector(goBackButtonPressed(tapGestureRecognizer:)))
+        goBackButton.addGestureRecognizer(tapGestureRecognizer2)
+        
+        recentSearchesTableView.dataSource = self
+        recentSearchesTableView.register(UINib(nibName: Constants.NibNames.recentSearch, bundle: nil), forCellReuseIdentifier: Constants.TableCellsIdentifier.recentSearch)
+    }
+
+    
+    @objc func goBackButtonPressed(tapGestureRecognizer: UITapGestureRecognizer) {
+        navigationController?.popViewController(animated: true)
+    }
+    
+
+    @IBAction func clearRecentSearchesPressed(_ sender: UIButton) {
+        print("Clear the table view and history in database")
+        recentSearchesArray = []
+        recentSearchesTableView.reloadData()
+    }
+
+    
+    func displaySearchResults(){
```

Comment:
Are you crating a new search results view every time you heat the search button?

Try to find a way just to update the data inside the search view and not adding a new view.
If you heat 100 times on the search button there will be 100 view one on top (view.addSubview(resultsView))?

Reply:
Not relevant on the updated version (as function no longer exists) - now have a results-table that starts hidden. When we have some data to display - we fill the table's array and set   table.isHidden = false
@abenartz